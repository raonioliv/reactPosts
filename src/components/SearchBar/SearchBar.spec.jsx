import { render, screen } from '@testing-library/react';
import { SearchBar } from '.';
import userEvent from '@testing-library/user-event';
describe('<SearchBar />', () => {
  it('should have the value of searchValue', () => {
    const fn = jest.fn();
    render(<SearchBar handleChange={fn} searchValue={'testando'} />);

    const value = 'any value';
    const input = screen.getByPlaceholderText(/Pesquisar/i);

    expect(input).toBeInTheDocument();
    userEvent.type(input, value);
    expect(input.value).toBe('testando');
  });

  it('should call handleChange at each keypress', () => {
    const fn = jest.fn();
    const value = 'any value';
    render(<SearchBar handleChange={fn} searchValue={value} />);
    const input = screen.getByPlaceholderText(/Pesquisar/i);

    userEvent.type(input, value);

    expect(input.value).toBe(value);
    expect(fn).toHaveBeenCalledTimes(value.length);
  });

  it('should match the snapshot', () => {
    const fn = jest.fn();
    const value = '';
    const { container } = render(<SearchBar handleChange={fn} searchValue={value} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
