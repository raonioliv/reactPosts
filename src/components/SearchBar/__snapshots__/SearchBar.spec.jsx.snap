// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> should match the snapshot 1`] = `
<div
  class="searchBar__wrapper inputWithIcon"
>
  <label
    class="searchBar__label"
    for="searchBar"
  >
    Pesquisar por posts
  </label>
  <input
    class="searchBar__input"
    id="searchBar"
    name="searchBar"
    placeholder="Pesquisar"
    title="Barra de pesquisa"
    type="text"
    value=""
  />
  <i
    class="search-icon searchBar__icon"
    role="button"
  />
</div>
`;
